sensor/common/registry/tlscheck_cache_test.go

```diff
@@ -0,0 +1,111 @@
+package registry
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCheckTLS(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("secure", func(t *testing.T) {
+		c := newTLSCheckCache(alwaysSecureCheckTLS)
+		secure, skip, err := c.CheckTLS(ctx, "fake")
+		assert.True(t, secure)
+		assert.False(t, skip)
+		assert.NoError(t, err)
+		assert.Len(t, c.results.GetAll(), 1)
+
+		// Ensure the results do not change when attempted again / using cache
+		secure, skip, err = c.CheckTLS(ctx, "fake")
+		assert.True(t, secure)
+		assert.False(t, skip)
+		assert.NoError(t, err)
+		assert.Len(t, c.results.GetAll(), 1)
+	})
+
+	t.Run("insecure", func(t *testing.T) {
+		c := newTLSCheckCache(alwaysInsecureCheckTLS)
+		secure, skip, err := c.CheckTLS(ctx, "fake")
+		assert.False(t, secure)
+		assert.False(t, skip)
+		assert.NoError(t, err)
+		assert.Len(t, c.results.GetAll(), 1)
+
+		// Ensure the results do not change when attempted again / using cache
+		secure, skip, err = c.CheckTLS(ctx, "fake")
+		assert.False(t, secure)
+		assert.False(t, skip)
+		assert.NoError(t, err)
+		assert.Len(t, c.results.GetAll(), 1)
+	})
+
+	t.Run("error", func(t *testing.T) {
+		c := newTLSCheckCache(alwaysFailCheckTLS)
+		secure, skip, err := c.CheckTLS(ctx, "fake")
+		assert.False(t, secure)
+		assert.False(t, skip)
+		assert.Error(t, err)
+		assert.Len(t, c.results.GetAll(), 1)
+
+		// Results expected to change, skip should be true due to previous error.
+		secure, skip, err = c.CheckTLS(ctx, "fake")
+		assert.False(t, secure)
+		assert.True(t, skip)
+		assert.NoError(t, err)
+		assert.Len(t, c.results.GetAll(), 1)
+	})
+}
+
+func TestAysncCheckTLS(t *testing.T) {
+	callCounts := map[string]int{}
+	callCountsMutex := sync.Mutex{}
+
+	countingCheckTLSFunc := func(_ context.Context, registry string) (bool, error) {
+		callCountsMutex.Lock()
+		defer callCountsMutex.Unlock()
+		callCounts[registry]++
+		return true, nil
+	}
+	regs := []string{"reg1", "reg2", "reg3"}
+
+	c := newTLSCheckCache(countingCheckTLSFunc)
```

Comment:
nit: may want to consider explicitly setting the TTL env var in case it were to change. Not terribly important, though

Reply:
Going to skip this :) - would require moving the package level variable

https://github.com/stackrox/stackrox/blob/ed5129b074988a360f8a7f566509a3ce90183d17/sensor/common/registry/tlscheck_cache.go#L21-L23

into the `newTLSCheckCache` - and I highly doubt the TTL will be lowered to sub 1 second necessary to affect the unit test